### ISSUE TITLE
[3.15.0] The native_constant_invocation rule removes trailing `\` on stdClass

### ISSUE THREAD

**Issue by HedicGuibert** [PHP-CS-Fixer/PHP-CS-Fixer#6832]
## Bug report

After upgrading to 3.15.0, the fixer removes the trailing `\` on `stdClass` if it used with an union type and the arg is passed by reference.

### Description

If an argument uses an union type and is passed by reference, the `native_constant_invocation` rule removes the trailing `\` on the `stdClass` type declaration. We can also see the issue with the spaces that is being reported in #6828 as well.

```shell
vendor/bin/php-cs-fixer fix src --verbose --dry-run --diff

1) src/Expand/Expander.php (native_constant_invocation)
-    private function handleNullPropertyAndGraphProperty(\stdClass|array &$result): bool
+    private function handleNullPropertyAndGraphProperty(stdClass | array &$result): bool
```

I tried removing the `&` on `$result` and the fixer stopped removing the `\`. The same class, as well as the whole project, has multiple `\stdClass|array` and none of them is having the same issue.
Setting the `native_constant_invocation` rule to false also fixes the issue.

### Runtime version

```
PHP CS Fixer 3.15.0 BoY42 by Fabien Potencier and Dariusz Ruminski.
PHP runtime: 8.2.3
```

### Configuration file

https://gist.github.com/HedicGuibert/30437a38ba306208c965c221e62979bd

**Comment by Wirone** [comment]
Could be related to #6828 and fixed in #6826. Could you check it somehow?

**Comment by HedicGuibert** [comment]
Oh my bad I didn't check #6826, after testing it it seems to fix the issue, I'm closing this.
